torch/_inductor/ir.py

```diff
@@ -6561,7 +6561,10 @@ def codegen(self, wrapper) -> None:  # type: ignore[no-untyped-def]
     def __init__(self, layout, input, indices: List[Tuple[Any, ...]]) -> None:  # type: ignore[no-untyped-def]
         super().__init__(None, layout, [input], ())
         self.name = V.graph.register_buffer(self)
-        V.graph.register_operation(self)
+        if input is not tuple or input is not list:
```

Comment:
I didn't register a buffer here to avoid register a redundant buffer, which introduces the redundant assign mentioned in the issue.